vllm/sequence.py

```diff
@@ -122,6 +126,36 @@ def get_output_len(self) -> int:
     def get_token_ids(self) -> List[int]:
         return self.prompt_token_ids + self.output_token_ids
 
+    def advance_prefill_range(self, size: int) -> int:
+        """Advance the prefill range by the specified amount
+
+        Args:
+            size: The amount to advance the prefill range.
+        Returns:
+            The actual number of advanced tokens.
+        """
+        self._prefill_start = self._prefill_end
+        # The increased range could be larger than the seq length.
+        # Clamp it to the seq length.
+        # Note that we use prompt_len + output_len instead of
+        # prompt_len here. This is because during recompute
+        # we need to prefill for both prompt and output.
+        self._prefill_end = min(self._prefill_end + size, self.get_len())
+        return self._prefill_end - self._prefill_start
+
+    def get_prefill_range(self) -> Tuple[int, int]:
+        """Returns the prefill range."""
+        return self._prefill_start, self._prefill_end
+
+    def get_num_unprefilled(self) -> int:
```

Comment:
Nit: can we use a more descriptive name?

Reply:
What about get_num_uncomputed_tokens?